Assignment 2 Yixuan Zhu.Rmd

```diff
@@ -0,0 +1,190 @@
+
+plot(g,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, edge.width=EDGE$count)
+
+````
+
+## Part II
+
+In Part II your task is to [look up](http://igraph.org/r/) in the igraph documentation and create a graph that sizes the student vertices in terms of the number of disciplinary actions they have recieved, and the teachers in terms of the number of disciplinary actions they have given out.
+
+```{r}
+
+V2.TCH <- EDGE %>% group_by(from) %>% summarise(sum(count))
+names(V2.TCH) = c("character","count")
+
+V2.STD <- EDGE %>% group_by(to) %>% summarise(sum(count))
+names(V2.STD) = c("character","count")
+
+
+VERTEX2 <- bind_rows(V2.TCH, V2.STD)
+g2 <- graph.data.frame(EDGE, directed=TRUE, vertices=VERTEX2)
+
+
+plot(g2,layout=layout.fruchterman.reingold)
+plot(g2,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender)
+
+##Use*5 to emphasize the size difference
+
+plot(g2,layout=layout.fruchterman.reingold, vertex.color=VERTEX$gender, vertex.size = (VERTEX2$count)*5)
+
+```
+
+
+
+
+## Part III
+
+Now practice with data from our class. Please create a **person-network** with the data set hudk4050-classes.csv. To create this network you will need to create a person-class matrix using the tidyr functions and then create a person-person matrix using `t()`. You will then need to plot a matrix rather than a data frame using igraph.
+
+
+```{r}
+D3 <- read.csv("hudk4050-classes.csv")
+library(dplyr)
+library(tidyr)
+
+##Combine name
+D3 <- D3 %>% unite(Name, First.Name, Last.Name, remove = TRUE, sep=" ")
+D3<-D3[-1,]
+
+##Gather
+D4<-gather(D3,Name,Class)
+D4[,1]<-D3$Name
+
+##Tidy
+D4$Class<-gsub(pattern = "\\s",replacement = "",D4$Class)
+cnt<-ifelse(D4$Class=="",D4$count<-0,D4$count<-1)
+D4$count<-cnt
+classes<-unique(D4$Class)
+D4[34,2]<-"HUDK4050"
+D4[87,2]<-"HUDM4125"
+D4[140,2]<-"HUDM5026"
+D4[193,2]<-"HUDM5126"
+D5<-D4[!D4$Class=="",]
+D5<-spread(D5,Class,count,fill=0)
+D5<-select(D5,-HUDK4050)
+
+##Matrix
+m1<-as.matrix(D5)
+row.names(m1)<-D5$Name
+m1<-m1[,-1]
+m1<-apply(m1,2,as.numeric)
+m2<-m1%*%t(m1)
+row.names(m2)<-D5$Name
+m1<-m1[,-1]
+m1<-apply(m1,2,as.numeric)
+m2<-m1%*%t(m1)
+row.names(m2)<-D5$Name
+colnames(m2)<-D5$Name
+diag(m2)<-NA
+
+##Graph
+library(igraph)
+m<-graph.adjacency(m2,mode="undirected")
+library(RColorBrewer)
+V(m)$cex<-0.5
+plot(m,layout=layout.fruchterman.reingold,vertex.color=rainbow(7),vertex.label.cex=0.4)
+```
+Once you have done this, also [look up](http://igraph.org/r/) how to generate the following network metrics: betweeness centrality and dregree. **Who is the most central person in the network?**
+
+
+```{r}
+##Betweenness Centrality
+betweeness_centrality <- betweenness(m)
+betweeness_centrality[betweeness_centrality == max(betweeness_centrality)]
+##Degree
+degrees <- degree(m)
+degrees[degrees == max(degrees)]
+
+##Yujun Zhang and Lintong Li are the most central people!
```

Comment:
Can you explain the difference between the two centrality measures? Which would you chose in this situation?